// uview-ui/index.js
// 看到此报错，是因为没有配置vue.config.js的【transpileDependencies】，详见：https://www.uviewui.com/components/npmSetting.html#_5-cli模式额外配置
const pleaseSetTranspileDependencies = {}, babelTest = pleaseSetTranspileDependencies?.test



// 引入全局mixin
import mixin from './libs/mixin/mixin.js'
// 小程序特有的mixin
import mpMixin from './libs/mixin/mpMixin.js'
// 全局挂载引入http相关请求拦截插件
import Request from './libs/luch-request'

// 路由封装
import route from './libs/util/route.js'
// 颜色渐变相关,colorGradient-颜色渐变,hexToRgb-十六进制颜色转rgb颜色,rgbToHex-rgb转十六进制
import colorGradient from './libs/function/colorGradient.js'

// 规则检验
import test from './libs/function/test.js'
// 防抖方法
import debounce from './libs/function/debounce.js'
// 节流方法
import throttle from './libs/function/throttle.js'
// 公共文件写入的方法
import index from './libs/function/index.js'

// 配置信息
import config from './libs/config/config.js'
// props配置信息
import props from './libs/config/props.js'
// 各个需要fixed的地方的z-index配置文件
import zIndex from './libs/config/zIndex.js'
// 关于颜色的配置，特殊场景使用
import color from './libs/config/color.js'
// 平台
import platform from './libs/function/platform'

const $u = {
    route,
    date: index.timeFormat, // 另名date
    colorGradient: colorGradient.colorGradient,
    hexToRgb: colorGradient.hexToRgb,
    rgbToHex: colorGradient.rgbToHex,
    colorToRgba: colorGradient.colorToRgba,
    test,
    type: ['primary', 'success', 'error', 'warning', 'info'],
    http: new Request(),
    config, // uView配置信息相关，比如版本号
    zIndex,
    debounce,
    throttle,
    mixin,
    mpMixin,
    props,
    ...index,
    color,
    platform
}

// 确保 window.uni 存在
if (typeof window !== 'undefined') {
    if (!window.uni) {
        window.uni = {}
    }
    // $u挂载到uni对象上
    window.uni.$u = $u
}

// 创建 Vue 3 兼容的过滤器
const createFilters = (app) => {
    // 时间格式化过滤器
    app.config.globalProperties.$filters = {
        timeFormat: (timestamp, format) => $u.timeFormat(timestamp, format),
        date: (timestamp, format) => $u.timeFormat(timestamp, format),
        timeFrom: (timestamp, format) => $u.timeFrom(timestamp, format)
    }
}

const install = (app) => {
    // Vue 3 不再支持过滤器，使用自定义的全局方法替代
    createFilters(app)
    
    // 挂载到 Vue 3 的全局属性中
    app.config.globalProperties.$u = $u
    
    // 全局混入
    app.mixin(mixin)
}

export default {
    install
}
